engineio/src/asynchronous/async_socket.rs

```diff
@@ -83,37 +81,45 @@ impl Socket {
         Ok(())
     }
 
-    /// Polls for next payload
-    pub(crate) async fn poll(&self) -> Result<Option<Packet>> {
-        loop {
-            if self.connected.load(Ordering::Acquire) {
-                if self.remaining_packets.read().await.is_some() {
-                    // SAFETY: checked is some above
-                    let mut iter = self.remaining_packets.write().await;
-                    let iter = iter.as_mut().unwrap();
-                    if let Some(packet) = iter.next() {
-                        return Ok(Some(packet));
-                    }
-                }
+    /// Creates a stream over the incoming packets, uses the streams provided by the
+    /// underlying transport types.
+    pub(crate) async fn stream(
```

Comment:
suggestion: does this need to be async?

Reply:
At the moment this is async as the address of a transport is locked with an async lock. If I switch to sync locking this will be fine, but blocking...